### PR TITLE
[FIX] Display first message when taking Livechat inquiry

### LINKED ISSUE #13668: When clicking on incoming livechat, message is blank
When Livechat Routing (Administration -> Livechat -> Routing) is set to Guest Pool, and there is an incoming livechat and you click on it

image

The message is coming as blank:

image

### PR DESCRIPTION
**Pull request by renatobecker**
Closes #13668.

Since the new Livechat client gets a new room before sending messages, this change has impacted on the `Guest Pool` routing method because now the first message is no longer stored into the `Inquiry` collection.
Due to that, now the `Take Inquiry` dialog will retrieve the first message by calling a server-side method.

### CODE REVIEW

File: app/livechat/client/views/sideNav/livechat.js [hunk 1]
```diff
@@ -113,21 +115,28 @@ Template.livechat.events({
...
+			if (error) {
+				return handleError(error);
 			}
+
+			modal.open({
+				title: t('Livechat_Take_Confirm'),
+				text: `${ t('Message') }: ${ result && result.msg }`,
+				showCancelButton: true,
+				confirmButtonColor: '#3085d6',
+				cancelButtonColor: '#d33',
```
Comment (tassoevan) [inline app/livechat/client/views/sideNav/livechat.js]: I guess it's safe to remove these color definitions.

File: app/livechat/client/views/sideNav/livechat.js [hunk 2]
```diff
@@ -113,21 +115,28 @@ Template.livechat.events({
...
-			confirmButtonText: t('Take_it'),
-		}, (isConfirm) => {
-			if (isConfirm) {
-				Meteor.call('livechat:takeInquiry', this._id, (error, result) => {
-					if (!error) {
-						roomTypes.openRouteLink(result.t, result);
-					}
-				});
+		const { rid } = this;
+		Meteor.call('livechat:getFirstRoomMessage', { rid }, (error, result) => {
```
Comment (tassoevan) [inline app/livechat/client/views/sideNav/livechat.js]: Why take the first message instead of the last one?

Comment (renatobecker) [reply app/livechat/client/views/sideNav/livechat.js]: @tassoevan, actually this change keeps the standard implementation of taking livechat inquiries.
Livechat users often send their first question when they start a conversation, so the last one may be out of context.

File: app/livechat/server/methods/getFirstRoomMessage.js [hunk 3]
```diff
@@ -0,0 +1,17 @@
+import { Meteor } from 'meteor/meteor';
+import { check } from 'meteor/check';
+import { Rooms, Messages } from '../../../models';
+
+Meteor.methods({
+	'livechat:getFirstRoomMessage'({ rid }) {
```
Comment (sampaiodiego) [inline app/livechat/server/methods/getFirstRoomMessage.js]: please add some permission check.. at least for logged in users.

Comment (renatobecker) [reply app/livechat/server/methods/getFirstRoomMessage.js]: @sampaiodiego, done.

Thanks.

**Comment by renatobecker** [comment]
> should it be fixed the other way around? like, fix by adding the message on the inquiry again?
> 
> what happened to that code? is it obsolete now? should it be removed then?

The new livechat gets the `room` before sending the first message, so when the inquiry is created, no message object is passed to the method.
But, since we have plans to change the behavior of taking livechat inquiries, it's possible that storing the first message will be necessary anymore.

**Pull request merged**